### ISSUE TITLE
`TF_MAKE_PP_{ARRAY,LIST,SEQ}` and `TF_PP_TUPLE_{SIZE,TO_LIST,TO_SEQ}` are unused in `pxr` and introduce boost dependencies

### ISSUE THREAD

**Issue by nvmkuruc** [PixarAnimationStudios/USD#2225]
### Description of Issue
The following macros in `preprocessorUtils.h` are unused in `pxr` and introduce additional dependencies on `BOOST_PP_EMPTY`, `BOOST_PP_TUPLE_TO_SEQ`, and `BOOST_PP_TUPLE_TO_LIST`.

```
#define TF_PP_TUPLE_SIZE(tuple) \
#define TF_PP_TUPLE_TO_LIST(tuple) \
#define TF_PP_TUPLE_TO_SEQ(tuple) \
#define TF_MAKE_PP_ARRAY(...) \
#define TF_MAKE_PP_LIST(...) \
#define TF_MAKE_PP_SEQ(...) \
```

### Steps to Reproduce
N/A

### System Information (OS, Hardware)
N/A

### Package Versions
N/A

### Build Flags
N/A

**Comment by spiffmon** [comment]
The TF_MAKE_XXX macros are, in fact, unused in our entire codebase.  The TF_PP_TUPLE_TO_XXX macros, however, are used in execution libraries that are likely to become part of OpenExec, so it's not immediately clear that moving them out of pxr now would be a win...

**Comment by tallytalwar** [comment]
Filed as internal issue #USD-7953

**Comment by nvmkuruc** [comment]
> The TF_MAKE_XXX macros are, in fact, unused in our entire codebase. The TF_PP_TUPLE_TO_XXX macros, however, are used in execution libraries that are likely to become part of OpenExec, so it's not immediately clear that moving them out of pxr now would be a win...

It's definitely something that can be waited on until the other preprocessor usage is addressed, but I might argue that there would still be a benefit to having a boost-free USD even if OpenExec wasn't.

**Comment by spiffmon** [comment]
> > The TF_MAKE_XXX macros are, in fact, unused in our entire codebase. The TF_PP_TUPLE_TO_XXX macros, however, are used in execution libraries that are likely to become part of OpenExec, so it's not immediately clear that moving them out of pxr now would be a win...
> 
> It's definitely something that can be waited on until the other preprocessor usage is addressed, but I might argue that there would still be a benefit to having a boost-free USD even if OpenExec wasn't.

An excellent point, @nvmkuruc , given that OpenExec is a ways out.

**Comment by nvmkuruc** [comment]
`TF_NUM_ARGS` can be added to the list of unused macros with the integration of #2530

**Comment by nvmkuruc** [comment]
@jesschimein @spiffmon @tallytalwar I think this can be closed now with 24.05.
